crates/sui-core/src/consensus_adapter.rs

```diff
@@ -226,48 +256,82 @@ impl ConsensusAdapter {
     }
 
     fn await_submit_delay(
+        &self,
         committee: &Committee,
         ourselves: &AuthorityName,
         transaction: &ConsensusTransaction,
     ) -> (impl Future<Output = ()>, usize) {
-        let (duration, position) = Self::submit_delay(committee, ourselves, transaction);
+        let (duration, position) = match &transaction.kind {
+            ConsensusTransactionKind::UserTransaction(certificate) => {
+                let tx_digest = certificate.digest();
+                let position = self.submission_position(committee, ourselves, tx_digest);
+                // DELAY_STEP is chosen as 1.5 * mean consensus delay
+                const DELAY_STEP: Duration = Duration::from_secs(7);
+                const MAX_DELAY_MUL: usize = 10;
+                (
+                    DELAY_STEP * std::cmp::min(position, MAX_DELAY_MUL) as u32,
+                    position,
+                )
+            }
+            _ => (Duration::ZERO, 0),
+        };
         (tokio::time::sleep(duration), position)
     }
 
-    fn submit_delay(
-        committee: &Committee,
-        ourselves: &AuthorityName,
-        transaction: &ConsensusTransaction,
-    ) -> (Duration, usize) {
-        if let ConsensusTransactionKind::UserTransaction(certificate) = &transaction.kind {
-            Self::submit_delay_certificate(committee, ourselves, certificate.digest())
-        } else {
-            (Duration::ZERO, 0)
-        }
-    }
-
     /// Check when this authority should submit the certificate to consensus.
     /// This sorts all authorities based on pseudo-random distribution derived from transaction hash.
-    /// Authorities higher in the list wait less time.
     ///
-    /// The function targets having only 1 consensus transaction submitted per user transaction
+    /// The function targets having 1 or 2 consensus transaction submitted per user transaction
```

Comment:
Why did you change this comment to `1 or 2`? I think logic in this PR still selects one validator only?

Reply:
In the case of a network partition, it will select 2

Reply:
Can you explain this? I see the code and only one validator receives position 0, where does second transactio comes from in normal case?
Mingwei made a change in testnet branch that would result in multiple validators submitting transaction, but this is not part of main branch or this PR